fiatconnect-api.md

```diff
@@ -391,105 +407,204 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
 
-The semantics of these claims are below:
+* A server MUST verify that the `Version` and `Chain ID` lines in the signed message are exactly `1` and `42220`, respectively.
 
-##### 3.2.1.2.1. `"sub"` Claim
+* If the `issued-at` field in the signed message is after the server's current timestamp, or if the `issued-at` field is after the `expiration-time` field, the server
+  MUST fail the request. If the server's current timestamp is past the `expiration-time` field, the server MUST fail the request. If the `expiration-time` field is
+  greater than four hours (14400 seconds) ahead of the `issued-at` field, the server MUST fail the request.
 
-The `sub`, or *subject* claim represents the user's Celo network address. This is a required claim. If it is missing, the server MUST respond
-to the client with an HTTP `400` error. This claim SHOULD be interpreted and used within server endpoints as a unique user identifier, and as the
-source of/destination for crypto funds during transfers. The address should be formatted as 42 hexadecimal characters prefixed by the string literal
-`0x`. If the `sub` claim does not match this format, the server MUST respond to the client with an HTTP `400` error.
+If all fields are acceptable, and the message is formatted correctly, the server MUST authenticate the user's session, and return an HTTP `200`.
+The server MUST attach the fields from the signed message to the session internally, in order to access data about the user and their session in other endpoints.
 
-##### 3.2.1.2.2. `"iss"` Claim
+Once a server authenticates a session using a particular nonce, the server MUST NOT allow any other sessions to be created using that nonce, until `expiration-time` of the session created
+using that nonce has been reached. The implementation of this is up to the server, but will require storing some global state about nonces in order to check if a client is attempting
+to sign-in using a nonce that is temporarily forbidden. Careful consideration should be taken to make sure that nonce management works correctly when scaling server resources.
 
-The `iss`, or *subject* claim represents the user's Celo address public key, dervied from their private key. This is a required claim.
-The public key MUST be in compressed form (see [here](https://docs.ethers.io/v5/api/utils/signing-key/) for more details), and exactly 33 bytes
-long.
+###### 3.2.1.5.3.2. Failure
 
-If it is missing, the server MUST respond to the client with an HTTP `400` error. The server MUST use this claim to verify the JWT signature.
-Since the address provided in the `"sub"` claim may not correspond to the public/private keypair used to sign and verify the JWT, the
-server MUST validate that the signature does indeed correspond to the address provided in the `"sub"` claim.
+On failure, the server MUST return an HTTP `401` error, along with an error code that details exactly why the request failed.
```

Comment:
Returning `401` instead of `400` seems like a good call, but does contradict what was said above. Also, saying that the MUST return a 4XX error code on failure would be broken by server errors which should return a 5XX error.

Reply:
Hm, not sure where I specify anything but a `401` in this section. And yeah, a `4xx` would be broken by `5xx`s, but that's true of every endpoint; not sure it's worth making an allowance for that explicitly in the spec.